### PR TITLE
Support automatic tag annotation

### LINKED ISSUE #3358: Add Tag from Automatic Annotation
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

How to tag an image with an automatic annotation?
I have a model deployed in nuclio and would like to return a tag if an element is, for example, not detected in an image.

### Current Behaviour
I tried to return a JSON stating that the type is a tag, but this results in an error.
My current workaround is to return a rectangle for the entire image, which states that the element is not in the image. 

### Expected Behaviour
Returning and adding a tag from the automatic annotation would be nice.

Example return JSON with a tag instead of a rectangle/polygon/point:
```
[
    {
        "confidence": "0.9999131",
        "label": "not found element",
        "type": "tag"
    }
]
```

### PR DESCRIPTION
**Pull request by subdavis**
### Motivation and context

resolves #3358 

### How has this been tested?

I have a full-frame classifier nuclio lambda and needed a way to save these tags.

* I ran the code in development using `docker-compose -f docker-compose.yml -f components/serverless/docker-compose.serverless.yml up -d`
* Invoke my nuclio classifier from the task summary page (full video) and from the annotator (single frame)
* See new tags appear in the UI.

I did see that the author of #3358 has included confidence values, which I do not believe are supported by tags. If I should support this somewhere, please LMK.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] ~~I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly~~ I do not see any documentation about what response types are supported for nuclio functions, so I don't think there's any documentation to update.  New documentation could be added about the general capabilities of this feature, but it seems out of scope for this PR.
- [ ] ~~I have added tests to cover my changes~~ I do not see tests related to this section of the code.  If they exist, please LMK and I'll add coverage.
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] ~~I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~ Not applicable

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] ~~I have updated the license header for each file (see an example below)~~ No new files added

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +40,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add a tutorial on attaching cloud storage AWS-S3 (<https://github.com/openvinotoolkit/cvat/pull/3745>)
   and Azure Blob Container (<https://github.com/openvinotoolkit/cvat/pull/3778>)
 - The feature to remove annotations in a specified range of frames (<https://github.com/openvinotoolkit/cvat/pull/3617>)
+- Add "tag" return type for automatic annotation in Nuclio (<https://github.com/openvinotoolkit/cvat/pull/3896>)
```
Comment (nmanovic) [inline CHANGELOG.md:43]: Could you please update the changelog? After we released 1.7.0 it isn't valid anymore.

File: cvat/apps/lambda_manager/views.py [hunk 2]
```diff
@@ -380,7 +383,17 @@ def reset(self):
 
             for anno in annotations:
                 label_id = labels.get(anno["label"])
-                if label_id is not None:
+                if label_id is None:
+                    continue # Invalid label provided
+                if anno["type"].lower() == "tag":
```
Comment (nmanovic) [inline cvat/apps/lambda_manager/views.py:388]: Could you please contribute any serverless function for classification? Thus we can test the PR.

**Comment by nmanovic** [comment]
@subdavis , please let me know if you have time to answer on my comments?

File: cvat/apps/lambda_manager/views.py (hunk 2 above)
Comment (subdavis) [reply cvat/apps/lambda_manager/views.py:388]: I don't think the introduction of a new serverless function is appropriately within scope for #3358

I actually don't even have one I can share: the code I have that generates full-frame detections is closed source.  I can provide some instructions about how to temporarily modify an existing function, but I don't really have time to write a new one.  I'm not really an ML developer.

The problem here seems to be one of documentation: the only way to understand the API between CVAT and Nuclio functions is to read this `views.py` file, so I think documentation would be ideal.

File: CHANGELOG.md (hunk 1 above)
Comment (subdavis) [reply CHANGELOG.md:43]: Yep, I can do that.

**Comment by subdavis** [comment]
Using any of the existing functions, you should be able to test by swapping `main` with this

``` python
import json

def init_context(context):
    pass

def handler(context, event):
    results = [{
        'type': 'tag',
        'label': 'example',
    }]
    return context.Response(body=json.dumps(results), headers={},
        content_type='application/json', status_code=200)

```

File: cvat/apps/lambda_manager/views.py (hunk 2 above)
Comment (nmanovic) [reply cvat/apps/lambda_manager/views.py:388]: @subdavis , it will be great to hear how in your company (kitware?) you are using CVAT to annotate data.

**Review by nmanovic — APPROVED** [review]
LGTM

File: cvat/apps/lambda_manager/views.py (hunk 2 above)
Comment (nmanovic) [reply cvat/apps/lambda_manager/views.py:388]: @subdavis , thanks for sharing the information.

**Pull request merged**